packages/ember/lib/index.js

```diff
@@ -212,19 +212,21 @@ Object.defineProperty(Ember, 'onerror', {
   @return {Object}
   @public
 */
-Ember.K = function K() {
-  deprecate(
-    'Ember.K is deprecated in favor of defining a function inline.',
-    false,
-    {
-      id: 'ember-metal.ember-k',
-      until: '3.0.0',
-      url: 'http://emberjs.com/deprecations/v2.x#toc_code-ember-k-code'
-    }
-  );
-
-  return this;
-};
+Object.defineProperty(Ember, 'K', {
+  get() {
+    deprecate(
+      'Ember.K is deprecated in favor of defining a function inline.',
+      false,
+      {
+        id: 'ember-metal.ember-k',
+        until: '3.0.0',
+        url: 'http://emberjs.com/deprecations/v2.x#toc_code-ember-k-code'
+      }
+    );
+
+    return function() { return this; };
```

Comment:
I think we should probably keep this as a stable function (in case folks are doing `this.something === Ember.K`).

Small tweak to support ^ would be:

```js
function deprecatedEmberK() { return this; }
Object.defineProperty(Ember, 'K', {
  get() {
    deprecate(/* current stuff here */);
    return deprecatedEmberK;
  }
});
```